### DISCUSSION TITLE
Loading not showing in preFetch hook

### DISCUSSION THREAD

**Discussion by mohamedelhosseiny** [quasarframework/quasar#10067] [CLI - SSR mode] [answered]
so this is my preFetch hook:
 ```
 async preFetch({ store }) {
    Loading.show();
    await Promise.all([store.dispatch("getLocationsProjects")]).then(
      success => {
        // Loading.hide();
      }
    );
  },
```
I know that prefetch on route enterance fetches from the server side, so `Loading` being a UI component will not be triggered. However, for ux sake I want to ask is there a way around this?

**Comment by justinbeaty** [comment, marked as answer]
@mohamedelhosseiny The prefetch runs on the server only the very first page load, which you may already know, but wanted to clarify. Does this not work on subsequent page navigations? Is it possible your store action is completing quicker than the default 500ms? Try:
```
Loading.show({
  delay: 0
})
```
